### PR TITLE
ELI5: Jekyll Collections

### LINKED ISSUE #207: Share Your Understanding of Jekyll Collections
Hey Ben,

I admire your lucid writing style and was wondering whether you'd be willing to apply that to Jekyll's [Collections](http://jekyllrb.com/docs/collections/) feature. You have been a champion of them since we introduced the feature last year. A few key questions:

- [What are collections? Explain to me like I'm 5.](http://www.reddit.com/r/Jekyll/comments/2op4ip/explain_collections_to_me_like_im_5/)
- How do they differ from pages?
- What should I use them for? Why do that over a straight pages solution?
- How do I configure them if, say, I want a different permalink instead of `/collection_name/doc_path`?

Use-cases will be especially helpful. Let me know what you think! Thank you. :smile:

### PR DESCRIPTION
**Pull request by benbalter**
Fixes #207.

@parkr still WIP, but thoughts?

### CODE REVIEW

File: _posts/2015-02-16-jekyll-collections.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+The tells Jekyll to look in the `_cupcakes` and `_cookies` folders for documents, and to read them into the appropriate collection, including [YAML front matter](http://jekyllrb.com/docs/frontmatter/), just as it would posts (but again, without the date).
+
+By default, collections are read in (and exposed as `site.[collection]`), but not included in the final site, at least not individually. If you wanted a page for each type of cupcake, you'd have to modify the `_config.yml` a bit:
+
+```yml
+collections:
+  cupcakes:
+    output: true
+    permalink: /cupcakes/:path/
```
Comment (parkr) [inline _posts/2015-02-16-jekyll-collections.md:62]: Add cookies here too with output false? Or with nil value.

File: _posts/2015-02-16-jekyll-collections.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+That way, `_cupcakes/chocolate.md` is outputted as `cupcakes/chocolate.html` when the site is built.
+
+
+ As I noted in [the original pitch](https://github.com/jekyll/jekyll/issues/1941),
```
Comment (parkr) [inline _posts/2015-02-16-jekyll-collections.md]: ?

**Comment by parkr** [comment]
Really great work so far :sparkles:

I like the simple example of the bakery, with a link out to the choosealicense site. I'd make a slightly stronger differentiation between posts (time-based) and collections (still grouped, but not chronological). You can say because they're not output by default, they're like data files on steroids-- front matter with arbitrary content.

So if I'm not a baker, what do I use this for? Projects for my resume, maybe? Doing more hacker-focused examples might help (just listing "if i'm making a site for this, i might have a collection called that"). 

Thank you for writing this!!!

We should add a 'sort_by' feature to collections... :)

**Comment by parkr** [comment]
The Ben Balter touch :tm: :raised_hands: :bow:

**Comment by benbalter** [comment]
@parkr thanks for (the idea and) all the feedback. Hope I got it all in there (at the very least, it pointed me in the right direction).  Hope it helps clear things up.
